pkg/didcomm/protocol/mediator/service.go

```diff
@@ -732,26 +746,34 @@ func (s *Service) setKeyUpdateResponseCh(msgID string, keyUpdateCh chan *Keylist
 	}
 }
 
-func (s *Service) getRouterConnectionID() (string, error) {
-	id, err := s.routeStore.Get(routeConnIDDataKey)
+func (s *Service) connectionExists(connID string) (bool, error) {
```

Comment:
This should be renamed to `ensureConnectionExists()` and should just return an error in case it doesn't (`ErrRouterNotRegistered`) or any generic IO error that may arise.